src/components/Bag/index.js

```diff
@@ -0,0 +1,540 @@
+
+    if (!menuIconClass) {
+        menuIconClass = classes.menuIcon;
+    }
+
+    // Convert the items into a map that the sharing dialog understands.
+    const sharingReducer = useCallback((acc, curr) => {
+        const newObj = { ...curr.item };
+
+        switch (newObj.type) {
+            case FILE_TYPE:
+            case FOLDER_TYPE:
+                acc[SHARING_TYPE.DATA] = [...acc.paths, newObj];
+                break;
+            case APP_TYPE:
+                acc[SHARING_TYPE.APPS] = [...acc.apps, newObj];
+                break;
+            case ANALYSIS_TYPE:
+                acc[SHARING_TYPE.ANALYSES] = [...acc.analyses, newObj];
+                break;
+            default:
+                break;
+        }
+        return acc;
+    }, []);
+
+    const convertItems = useCallback(
+        (data) => {
+            let converted = [];
+
+            if (data?.contents?.items) {
+                converted = data.contents.items.map((item) =>
+                    createNewBagItem(item)
+                );
+            }
+
+            setAllItems(converted);
+
+            setSharingResources(
+                converted.reduce(sharingReducer, defaultSharingResources())
+            );
+
+            setDownloadPaths(
+                converted
+                    .filter((item) => item.downloadable)
+                    .map((item) => item.path)
+            );
+        },
+        [sharingReducer]
+    );
+
+    const { isError: hasErrored, data, isLoading, error } = facade.useBag({
+        enabled: userProfile,
+        onSuccess: convertItems,
+    });
+
+    const removeItem = facade.useBagRemoveItem({
+        handleError: (error) => {
+            showErrorAnnouncer(t("removeItemError"), error);
+        },
+    });
+
+    const clearAll = facade.useBagRemoveItems({
+        handleError: (error) => {
+            showErrorAnnouncer(t("removeAllItemsError"), error);
+        },
+    });
+
+    useEffect(() => {
+        if (hasErrored) {
+            showErrorAnnouncer(t("fetchBagError"), error);
+        }
+    }, [hasErrored, error, showErrorAnnouncer, t]);
+
+    useEffect(() => {
+        convertItems(data);
+    }, [data, convertItems]);
+
+    useEffect(() => {
+        setBadgeCount(allItems.length);
+    }, [allItems, setBadgeCount]);
+
+    const handleSharingClick = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        setBagDlgOpen(false);
+        setSharingOpen(true);
+    };
+
+    const handleMenuClick = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        setBagDlgOpen(!bagDlgOpen);
+    };
+
+    const handleClose = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        setBagDlgOpen(false);
+    };
+
+    const handleDownloadClick = (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        setDownloadDlgOpen(true);
+        setBagDlgOpen(false);
+    };
```

Comment:
We should check if there are items for download before opening the download link dialog.

Reply:
Should I disable the button if there aren't any items to download? I know that we've been trying to avoid that, but it seems less annoying than having an error announcer pop up or just having nothing happen.

Reply:
Download button should not appear when there are no items that can be downloaded.

Reply:
This is fixed now.